template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveKnoxConfigProvider.java

```diff
@@ -23,17 +25,22 @@ public class HiveKnoxConfigProvider implements CmTemplateComponentConfigProvider
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject templatePreparationObject) {
         String filePerEvent = ConfigUtils.getSafetyValveProperty("hive.hook.proto.file.per.event", "true");
-
+        List<ApiClusterTemplateConfig> serviceConfigs = new ArrayList<>();
         String cdhVersion = templateProcessor.getVersion().orElse("");
         if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_1_0)) {
-            return List.of(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, filePerEvent));
+            serviceConfigs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, filePerEvent));
         } else {
             KerberosConfig kerberosConfigOpt = templatePreparationObject.getKerberosConfig().get();
             String realm = kerberosConfigOpt.getRealm();
             String principal = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.principal", "HTTP/_HOST@" + realm);
             String keytab = ConfigUtils.getSafetyValveProperty("hive.server2.authentication.spnego.keytab", "hive.keytab");
-            return List.of(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
+            serviceConfigs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, principal + keytab + filePerEvent));
+        }
+        if (CMRepositoryVersionUtil.isS3SslChannelModeSupported(cdhVersion, templatePreparationObject.getCloudPlatform())) {
+            String sslChannelMode = ConfigUtils.getSafetyValveProperty("fs.s3a.ssl.channel.mode", "openssl");
+            serviceConfigs.add(config(HIVE_SERVICE_CONFIG_SAFETY_VALVE, sslChannelMode));
```

Comment:
This will end up adding HIVE_SERVICE_CONFIG_SAFETY_VALVLE twice... Does that work? (As against adding sslChannelMode similar to `principal + keytab + filePerEvent`)

Reply:
These configurations are delta. They always get prefixed with existing configs from template. Please look at addServiceConfigs in CmTemplateProcessor.

Reply:
These configurations are delta. They always get prefixed with existing configs from template. Please look at addRoleConfigs in CmTemplateProcessor.